Document/0x05b-Basic-Security_Testing.md

```diff
@@ -774,24 +773,259 @@ Note that if you have the original source code and use Android Studio, you do no
+
+###### Other App Resources
+
+It is normally worth taking a look at the rest of the resources and files that you may find in the root folder of the APK as some times they contain additional goodies like key stores, encrypted databases, certificates, etc.
+
+##### Accessing App Data Directories
 
-##### Installed Apps
+Once you have installed the app, there is further information to explore, where tools like objection come in handy.
 
-##### App Basic Information
+When using objection you can retrieve different kinds of information, where `env` will show you all the directory information of the app.
 
-###### Sandbox
+```shell
+$ objection -g sg.vp.owasp_mobile.omtg_android explore
+
+...g.vp.owasp_mobile.omtg_android on (google: 8.1.0) [usb] # env
 
-###### Permissions
+Name                    Path
+----------------------  ---------------------------------------------------------------------------
+cacheDirectory          /data/user/0/sg.vp.owasp_mobile.omtg_android/cache
+codeCacheDirectory      /data/user/0/sg.vp.owasp_mobile.omtg_android/code_cache
+externalCacheDirectory  /storage/emulated/0/Android/data/sg.vp.owasp_mobile.omtg_android/cache
+filesDirectory          /data/user/0/sg.vp.owasp_mobile.omtg_android/files
+obbDir                  /storage/emulated/0/Android/obb/sg.vp.owasp_mobile.omtg_android
+packageCodePath         /data/app/sg.vp.owasp_mobile.omtg_android-kR0ovWl9eoU_yh0jPJ9caQ==/base.apk
+```
 
-###### Native Libs
+Among this information we find:
 
-###### ... other
+- The internal data directory (aka. sandbox directory) which is at `/data/data/[package-name]` or `/data/user/0/[package-name]`
+- The external data directory at `/storage/emulated/0/Android/data/[package-name]` or `/sdcard/Android/data/[package-name]`
+- The path to the app package in `/data/app/`
 
-##### Accessing App Data
+The internal data directory is used by the app to store data created during runtime and has the following basic structure:
+
+```shell
+...g.vp.owasp_mobile.omtg_android on (google: 8.1.0)  [usb] # ls
+Type       ...  Name
+---------  ...  -------------------
+Directory  ...  cache
+Directory  ...  code_cache
+Directory  ...  lib
+Directory  ...  shared_prefs
+Directory  ...  files
+Directory  ...  databases
+
+Readable: True  Writable: True
+```
+
+Each of the folders having its own purpose:
+
+- **cache**: This location is used for data caching. For example, the WebView cache is found in this directory.
+- **code_cache**: This is the location of the file system's application-specific cache directory designed for storing cached code. On devices running Lollipop or later Android versions, the system will delete any files stored in this location when the app or the entire platform is upgraded.
```

Comment:
Don't use "lolipop" or other code names. According to style guide, use this format (eg): Android 9 (API level 28)

Reply:
oh, I'll correct it. It was copy / pasted from the other section